src/apps/lwaftr/ipv6_apps.lua

```diff
@@ -147,3 +152,61 @@ function Fragmenter:push ()
       end
    end
 end
+
+function ICMPEcho:new(conf)
+   local addresses = {}
+   if conf.address then
+      addresses[ffi.string(conf.address, 16)] = true
+   end
+   if conf.addresses then
+      for _, v in ipairs(conf.addresses) do
+         addresses[ffi.string(v, 16)] = true
+      end
+   end
+   return setmetatable({addresses = addresses}, {__index = ICMPEcho})
```

Comment:
Here `address` is a `char[4]` cdata object, which won't convert directly to a Lua numeric value, hence the explicit use of `rd32()`.